common-ktx/src/main/java/co/nimblehq/common/extensions/DateExt.kt

```diff
@@ -0,0 +1,315 @@
+    get() = with(second = 0, millisecond = 0)
+
+/**
+ * Get the last time of the minute
+ */
+val Date.endOfMinute: Date
+    get() = with(second = 59, millisecond = 999)
+
+/**
+ * Convert date to specific string format
+ *
+ * @param formatPattern - format date, ex: dd/MM/yyyy
+ */
+fun Date.toString(formatPattern: String): String = SimpleDateFormat(formatPattern).format(this)
+
+/**
+ * Check if the current date is Sunday
+ */
+val Date.isSunday: Boolean
+    get() {
+        val calendar = Calendar.getInstance()
+        calendar.time = this
+        return calendar.get(Calendar.DAY_OF_WEEK) == Calendar.SUNDAY
+    }
+
+/**
+ * Check if the current date is Monday
+ */
+val Date.isMonday: Boolean
+    get() {
+        val calendar = Calendar.getInstance()
+        calendar.time = this
+        return calendar.get(Calendar.DAY_OF_WEEK) == Calendar.MONDAY
+    }
+
+/**
+ * Check if the current date is Tuesday
+ */
+val Date.isTuesday: Boolean
+    get() {
+        val calendar = Calendar.getInstance()
+        calendar.time = this
+        return calendar.get(Calendar.DAY_OF_WEEK) == Calendar.TUESDAY
+    }
+
+/**
+ * Check if the current date is Wednesday
+ */
+val Date.isWednesday: Boolean
+    get() {
+        val calendar = Calendar.getInstance()
+        calendar.time = this
+        return calendar.get(Calendar.DAY_OF_WEEK) == Calendar.WEDNESDAY
+    }
+
+/**
+ * Check if the current date is Thursday
+ */
+val Date.isThursday: Boolean
+    get() {
+        val calendar = Calendar.getInstance()
+        calendar.time = this
+        return calendar.get(Calendar.DAY_OF_WEEK) == Calendar.THURSDAY
+    }
+
+/**
+ * Check if the current date is Friday
+ */
+val Date.isFriday: Boolean
+    get() {
+        val calendar = Calendar.getInstance()
+        calendar.time = this
+        return calendar.get(Calendar.DAY_OF_WEEK) == Calendar.FRIDAY
+    }
+
+/**
+ * Check if the current date is Saturday
+ */
+val Date.isSaturday: Boolean
+    get() {
+        val calendar = Calendar.getInstance()
+        calendar.time = this
+        return calendar.get(Calendar.DAY_OF_WEEK) == Calendar.SATURDAY
+    }
+
+/**
+ * Check if the current date is Weekend
+ */
+val Date.isWeekend: Boolean
+    get() = this.isSaturday || this.isSunday
+
+/**
+ * Check if the current date is Weekday
+ */
+val Date.isWeekday: Boolean
+    get() = !this.isWeekend
+
+/**
+ * Check if the current date is Today
+ */
+val Date.isToday: Boolean
+    get() = isDateIn(this, 0)
+
+/**
+ * Check if the current date is Yesterday
+ */
+val Date.isYesterday: Boolean
+    get() = isDateIn(this, -1)
+
+/**
+ * Check if the current date is Tomorrow
+ */
+val Date.isTomorrow: Boolean
+    get() = isDateIn(this, 1)
+
+/**
+ * Check if the date is in a specific time
+ *
+ * @param date - Date want to check
+ * @param variable - time of date
+ *
+ * @return value to check if the current date is in expected date
+ */
+private fun isDateIn(date: Date, variable: Int = 0): Boolean {
```

Comment:
🤔 

> @param variable - time of date

I think we should avoid using `variable` for params and from the extension document: `time of date` doesn't make sense. Because from your use case, you are inputting 0,1,-1, not the time of date, right? 🤔

Reply:
@hoangnguyen92dn okay, let me change it.